applications/tari_validator_node/src/lib.rs

```diff
@@ -129,6 +130,11 @@ pub async fn run_validator_node_with_cli(config: &ApplicationConfig, cli: &Cli)
     );
 
     let (base_node_client, wallet_client) = create_base_layer_clients(config).await?;
+    let connectivity = initialize(node_identity.clone(), config, shutdown.to_signal())
```

Comment:
You're initializing comms twice. Rather create the `TariCommsValidatorNodeClientFactory` inside spawn_services. You can use `comms.connectivity()` to get the handle there.

Reply:
Nice ! Definitely, refactored the code accordingly :)